### PR TITLE
Les 452 user page

### PR DESCRIPTION
**Pull request by Kurilo-Pavel**

### CODE REVIEW

File: src/components/body/header/Avatar.tsx [hunk 1]
```diff
@@ -46,8 +50,8 @@ const Avatar = ({
       />
       <ul className="list list-user">
         <li className="user-li">
-          <Link to="/user/personalData" className="user-link">
-            <FormattedMessage id="app.avatar.personalData" />
+          <Link to="/user/userPage" className="user-link">
+            <FormattedMessage id="app.avatar.personalData"/>
```
Comment (Terapool) [inline src/components/body/header/Avatar.tsx:54]: [space]/&gt;

File: src/components/body/header/Avatar.tsx [hunk 2]
```diff
@@ -57,10 +61,10 @@ const Avatar = ({
             language={language}
             setLanguage={setLanguage}/>
         </li>
-        <li className="user-li">
-          <div className="log-out" onClick={signOut}>
-            <FormattedMessage id="app.avatar.logOut" />
-            <img src={Logout} alt="logout" />
+        <li className="user-li" onClick={signOut}>
+          <div className="log-out" >
+            <FormattedMessage id="app.avatar.logOut"/>
```
Comment (Terapool) [inline src/components/body/header/Avatar.tsx:66]: [space]/&gt;

File: src/components/body/header/Avatar.tsx [hunk 3]
```diff
@@ -57,10 +61,10 @@ const Avatar = ({
...
+            <img src={Logout} alt="logout"/>
```
Comment (Terapool) [inline src/components/body/header/Avatar.tsx:67]: [space]/&gt;

File: src/pages/FirstRegistrationForm.tsx [hunk 4]
```diff
@@ -39,68 +41,78 @@ const symbols = PASSWORD.symbols;
...
+  const [isRegEmail, setIsRegEmail] = useState<string | unknown>("");
   const initialValues: FormValues = {
     email: "",
     password: "",
     confirmPassword: "",
     hasTermsAndConditions: false
   };
 
+  useEffect(() => {
+    if (isRegEmail===false) {
```
Comment (Terapool) [inline src/pages/FirstRegistrationForm.tsx:55]: spaces:|
(isRegEmail[space]===[space]false)

File: src/pages/LoginPage.tsx [hunk 5]
```diff
@@ -34,7 +32,20 @@ const LoginPage = () => {
   const intl = useIntl();
   const dispatch = useAppDispatch();
   const navigate = useNavigate();
-  const isEmail = useAppSelector(state => state.login.isEmail);
+  const [value, setValue] = useState({email:"", password:""});
+  const [isLogEmail, setIslogEmail] = useState<string | unknown>("");
+
```
Comment (Terapool) [inline src/pages/LoginPage.tsx:37]: One empty line as separator is enough. No need in extra lines here...

File: src/pages/YourselfPage.tsx [hunk 6]
```diff
@@ -59,31 +59,24 @@ const YourselfPage = ({
...
-    }
-    if (!UserRegex.test(values.name)) {
+    if (UserRegex.test(values.name)) {
       errors.name = intl.formatMessage({id: "app.YourselfPage.errorIncorrectName"});
     }
     if (values.name.length === 0) {
       errors.name = intl.formatMessage({id: "app.YourselfPage.errorFieldEmpty"});
     }
-    if (values.name.length < USERNAME.minLength && values.name.length > 0) {
+    if (values.name.length < minSymbol && values.name.length > 0) {
```
Comment (Terapool) [inline src/pages/YourselfPage.tsx:73]: simply  values.name.length without &gt; 0

File: src/store/loginName/loginSlice.ts [hunk 7]
```diff
@@ -100,39 +105,106 @@ export const changePassword = createAsyncThunk(
...
+    const response = await requestApi(
+      `${process.env.REACT_APP_BACKEND_URL}/users/${items.name}`, "PUT", items.object);
+    const data = response.json();
+    return data;
+  }
+);
+
+export const editUserEmail = createAsyncThunk(
+  "user/editUserEmailStatus",
+  async (token:string)=>{
```
Comment (Terapool) [inline src/store/loginName/loginSlice.ts:132]: spaces

File: src/store/sign_up/signUpSlice.ts [hunk 8]
```diff
@@ -48,4 +54,5 @@ const userSlice = createSlice({
       state.user = action.payload;
     })
   }
-})
\ No newline at end of file
+})
+export default userSlice.reducer;
```
Comment (Terapool) [inline src/store/sign_up/signUpSlice.ts:58]: empty line at the file end

**Review by Terapool — CHANGES_REQUESTED** [review]
Please, see if you really need this file: 
src/components/icons/131.jpg

File: src/components/PasswordAndConfirm.tsx [hunk 9]
```diff
@@ -9,7 +9,7 @@ import {useState} from "react";
 type PasswordProps = {
   minSymbol: number;
   maxSymbol: number;
-  isConfirm: boolean;
+  isConfirm: boolean|string;
```
[the remark this answers is not in the document]
Comment (Kurilo-Pavel) [reply src/components/PasswordAndConfirm.tsx:12]: if isConfirm = true - it component for Confirm,

File: src/components/body/Body.tsx [hunk 10]
```diff
@@ -2,20 +2,20 @@ import "./Body.scss";
...
 
-const Body = ({ onLanguageSwitch }: BodyProps) => {
+const Body = ({onLanguageSwitch}: BodyProps) => {
   return (
     <div className="body-page">
-      <Header onLanguageSwitch={onLanguageSwitch} />
-      <NavbarStudyStudio menuType={NAVBAR_ADMIN} />
-      <Main />
-      <Footer />
+      <Header onLanguageSwitch={onLanguageSwitch}/>
```
[the remark this answers is not in the document]
Comment (Kurilo-Pavel) [reply src/components/body/Body.tsx:15]: space befor /> removed priter.

File: src/components/body/content/categories/CategoriesUser.tsx [hunk 11]
```diff
@@ -21,33 +22,33 @@ const CategoriesUser = () => {
...
-      description={obj.description} 
+      name={obj.name}
+      description={obj.description}
     />
   ));
 
   const skeleton = [...new Array(SKELETON_AMOUT)].map((_, index) =>
-    <SkeletonCategory key={index} />);
-  
+    <SkeletonCategory key={index}/>);
```
[the remark this answers is not in the document]
Comment (Kurilo-Pavel) [reply src/components/body/content/categories/CategoriesUser.tsx:47]: it remove pritter

**Review by NatashaMikhaltsova — APPROVED** [review]

**Review by valer-sky — APPROVED** [review]

**Review by Terapool — APPROVED** [review]

**Pull request merged**